client/src/components/WinLoss.svelte

```diff
@@ -0,0 +1,37 @@
+<script lang="ts">
+  import { verifyExpiration } from "$lib/firebase/firebase-functions";
+  import { GAME_STATE_DURATIONS, type Lobby } from "$lib/firebase/firestore-types/lobby";
+  import { onDestroy, onMount } from "svelte";
+  import { authStore as user } from "$stores/auth";
+
+  export let lobby: Lobby;
+  export let lobbyCode: string;
+  let countdown = GAME_STATE_DURATIONS.END;
+  let timer: ReturnType<typeof setInterval>;
+
+  onMount(() => {
+    timer = setInterval(() => {
+      if (lobby.expiration != undefined) {
+        const diff = Math.floor((lobby.expiration.toMillis() - Date.now()) / 1000);
+        countdown = diff;
+      }
+    }, 1000);
+  });
+  onDestroy(() => {
+    clearInterval(timer);
+  });
+
+  // apply stats
+  $: if (countdown == 0 && $user?.uid == lobby.uids[0]) {
+    clearInterval(timer);
+    verifyExpiration({ code: lobbyCode });
+  }
+  $: if (countdown <= -5) {
+    verifyExpiration({ code: lobbyCode });
+    clearInterval(timer);
```

Comment:
I don't like that one of these is clearInterval then verifyExpiration and the other is verifyExpiration then clearInterval